internal/contracts/manager.go

```diff
@@ -196,6 +202,31 @@ func (cm *contractManager) GetFFIs(ctx context.Context, filter ffapi.AndFilter)
 	return cm.database.GetFFIs(ctx, cm.namespace, filter)
 }
 
+func (cm *contractManager) verifyListeners(ctx context.Context) error {
+
+	var page uint64
+	var pageSize uint64 = 50
+	verifyCount := 0
+	for {
+		f := database.ContractListenerQueryFactory.NewFilterLimit(ctx, pageSize).And().Skip(page * pageSize)
+		listeners, _, err := cm.database.GetContractListeners(ctx, cm.namespace, f)
+		if err != nil {
+			return err
+		}
+		if len(listeners) == 0 {
+			log.L(ctx).Infof("Listener restore complete. Verified=%d", verifyCount)
+			return nil
+		}
+		for _, l := range listeners {
+			if err := cm.checkContractListenerExists(ctx, l); err != nil {
+				return err
+			}
+		}
```

Comment:
I think we need to increment `verifyCount` right here. Currently it doesn't look like it's ever updated.